src/main/java/teammates/ui/controller/InstructorFeedbackEditPageData.java

```diff
@@ -12,68 +12,297 @@
+                                                                     question.numberOfEntitiesToGiveFeedbackTo);
+        qnForm.setQuestionHasResponses(questionHasResponses.get(question.getId()));
+        
+        visibilitySettings.setVisibilityMessages(question.getVisibilityMessage());
+        
+        qnForm.setQuestionSpecificEditFormHtml(question.getQuestionDetails().getQuestionSpecificEditFormHtml(question.questionNumber));
+        qnForm.setEditable(false);
+        
+        qnForms.add(qnForm);
+    }
+
+
+    private void buildNewQuestionForm(FeedbackSessionAttributes feedbackSession,
+                                      List<FeedbackQuestionAttributes> questions) {
+        newQnForm = new FeedbackQuestionEditForm();
+        newQnForm.setDoneEditingLink(new Url(Const.ActionURIs.INSTRUCTOR_FEEDBACKS_PAGE)
+                                        .withUserId(account.googleId)
+                                        .withCourseId(feedbackSession.courseId)
+                                        .withSessionName(feedbackSession.feedbackSessionName));
+        
+        newQnForm.setAction(new Url(Const.ActionURIs.INSTRUCTOR_FEEDBACK_QUESTION_ADD));
+        newQnForm.setCourseId(feedbackSession.courseId);
+        newQnForm.setFeedbackSessionName(feedbackSession.feedbackSessionName);
+        newQnForm.setQuestionNumberSuffix("");
+        
+        newQnForm.setQuestionTypeOptions(getQuestionTypeChoiceOptions());
+      
+        FeedbackQuestionFeedbackPathSettings feedbackPathSettings = new FeedbackQuestionFeedbackPathSettings();
+        FeedbackQuestionVisibilitySettings visibilitySettings = new FeedbackQuestionVisibilitySettings();
+        
+        newQnForm.setFeedbackPathSettings(feedbackPathSettings);
+        newQnForm.setVisibilitySettings(visibilitySettings);
+        
+        feedbackPathSettings.setGiverParticipantOptions(getParticipantOptions(null, true));
+        feedbackPathSettings.setRecipientParticipantOptions(getParticipantOptions(null, false));
+        feedbackPathSettings.setNumOfEntitiesToGiveFeedbackToValue(1);
+        
+        newQnForm.setNumOfQuestionsOnPage(questions.size());
+        newQnForm.setQuestionNumberOptions(getQuestionNumberOptions(questions.size() + 1));
+      
+        newQnForm.setQuestionSpecificEditFormHtml(getNewQuestionSpecificEditFormHtml());
+        newQnForm.setEditable(true);
+        
+        setDefaultVisibilityOptions(visibilitySettings, feedbackPathSettings);
+    }
+
+
+    private void setDefaultVisibilityOptions(FeedbackQuestionVisibilitySettings visibilityOptions,
+                                             FeedbackQuestionFeedbackPathSettings feedbackPathSettings) {
+        Map<String, Boolean> isGiverNameVisible = new HashMap<String, Boolean>();
+        Map<String, Boolean> isRecipientNameVisible = new HashMap<String, Boolean>();
+        Map<String, Boolean> isResponsesVisible = new HashMap<String, Boolean>();
```

Comment:
Adding "For" to the end of these may help; their names seem like normal booleans right now, rather than maps.